### PR TITLE
blob: show code-intel tooltips only for languages supported

### PR DESCRIPTION
**Pull request by taras-yemets**
Partially addresses https://github.com/sourcegraph/sourcegraph/issues/47619

Do not show code-intel tooltips if language is not supported ("No hover info" message is always shown for such languages).

We still enable [`tokenSelectionExtension`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4e4ff62a3d7a60abb0616aba500d7b24508d672d/-/blob/client/web/src/repo/blob/codemirror/token-selection/extension.ts?L47-105&subtree=true) if language is not supported (despite extensions enabled by `tokenSelectionExtension` completely relying on [`Language['providers']`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ff466c55b67a09c03769a0e56a9677a99943850b/-/blob/client/shared/src/codeintel/legacy-extensions/providers.ts?L31-37), see [`findLanguageMatchingDocument`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ff466c55b67a09c03769a0e56a9677a99943850b/-/blob/client/shared/src/codeintel/api.ts?L155-163)) for selection-driven navigation and go to definition (resulting in "No definition found") to work.

https://user-images.githubusercontent.com/25318659/220310060-ef39b341-cdb1-4f5a-9c32-e4ef68812b0d.mov

## Test plan
- Ensure selection-based navigation and go to definition work for all languages (supported and not)
- Ensure that no code-intel tooltips are displayed only for [supported languages](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ff466c55b67a09c03769a0e56a9677a99943850b/-/blob/client/shared/src/codeintel/api.ts?L155-163)

## App preview:

- [Web](https://sg-web-taras-yemets-cm-blob-do-not-init.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.15 kb) | 0.00% (+0.15 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/taras-yemets/cm-blob-do-not-init-token-selection-ext-if-lang-not-supported/compare-report.html;tab=live_object) for a full comparison between the commits 9b6b570 and 74907f0 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by olafurpg — APPROVED** [review]

**Comment by taras-yemets** [comment]
@olafurpg, I marked PR as draft again as I had doubts if we really want to disable all the election-driven nav extensions for files with not supported language.
I this case, keyboard nav, go to the definition with "no definition found" and some other features will stop working for these files (e.g., keyboard navigation currently works for [app.scss](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4e4ff62/-/blob/client/browser/src/app.scss?subtree=true), but will stop working with the changes introduced by this PR). 
What do you think about it? Should we disable only hover tooltips for not supported languages as you initially suggested (https://github.com/sourcegraph/sourcegraph/issues/47619#issuecomment-1431245422)?

File: client/web/src/repo/blob/codemirror/token-selection/code-intel-tooltips.ts [hunk 1]
```diff
@@ -79,6 +81,9 @@ export const codeIntelTooltipsState = StateField.define<Record<CodeIntelTooltipT
                         return acc
                     }, [] as NonNullable<CodeIntelTooltipState>[])
                     .map(({ tooltip }) => tooltip)
+                    .filter(tooltip =>
+                        tooltip instanceof CodeIntelTooltip ? (isLanguageSupported ? tooltip : null) : tooltip
```
Comment (philipp-spiess) [inline client/web/src/repo/blob/codemirror/token-selection/code-intel-tooltips.ts:85]: I’m curious why we need the instanceof check here, what other tooltips are passed through the `codeIntelTooltipsState`?

**Review by philipp-spiess — APPROVED** [review]

File: client/web/src/repo/blob/codemirror/token-selection/code-intel-tooltips.ts (hunk 1 above)
Comment (taras-yemets) [reply client/web/src/repo/blob/codemirror/token-selection/code-intel-tooltips.ts:85]: [`LoadingTooltip`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4ba9c8c3058a8776486241e2540a971ff3957993/-/blob/client/web/src/repo/blob/codemirror/tooltips/LoadingTooltip.ts?L3:14-3:28#tab=references) and [`TemporaryTooltip`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4ba9c8c3058a8776486241e2540a971ff3957993/-/blob/client/web/src/repo/blob/codemirror/tooltips/TemporaryTooltip.tsx?L27:17-27:37#tab=references) can also be in state.

**Pull request merged**